pcdsutils/log.py

```diff
@@ -888,3 +888,36 @@ def should_demote(
             for implementing this method.
         """
         return 'Subscription %s callback exception' in info.message
+
+
+class PydmDemotionFilter(DemotionFilter):
+    """
+    Demote verbose logs from pydm, particulary those that come at exit.
+
+    Currently, this demotes the "Unable to remove connection" logger exit
+    spam from "ERROR" to "DEBUG".
+
+    These can be caused by some poorly understood race conditions at program
+    close and are not useful for the end user.
+    """
+    default_logger = logging.getLogger("pydm.widgets.channel")
+
+    def should_demote(
+        self,
+        record: logging.LogRecord,
+        info: RecordInfo,
+    ) -> bool:
+        """
+        Return True if we should demote the record, False otherwise.
+
+        Demote if we see the "Unable to remove connection" error.
+
+        Parameters
+        ----------
+        record: logging.LogRecord
+            The record instance passed into the filter.
+        info: RecordInfo
+            The record info dataclass, which may be more convenient
+            for implementing this method.
+        """
+        return "Unable to remove connection" in info.message
```

Comment:
My only thought here is that we might also filter based on the type exception thrown (`RuntimeError: wrapped C/C++ object of type Connection has been deleted`).  I don't know of any other case where we are unable to remove a connection, but I imagine that might be important to note.  

Though I suppose this only demotes the exceptions instead of removing them, so it's probably not necessary to be so specific.

Reply:
This is worth pondering a bit

Reply:
I think I want to leave this as-is, but not because I'm sure that it's best. I don't know which way is best and I think as-is is good enough.

I'd be hesitant to make this more specific because any kind of exception in this block will be a similarly-sized traceback regardless of the type, and all of them will be irrelevant to the user who isn't in the business of creating/deleting widgets and opening/closing connections.

I'm wondering a bit if there are other message that would be helpful to target pre-emptively here- but it's hard to predict these things.

Reply:
I think this is a fair assessment.  I was on the fence about it as well, so I'm not bothered either way